eip712/src/main/java/pm/gnosis/eip712/EIP712JsonParser.kt

```diff
@@ -0,0 +1,83 @@
+package pm.gnosis.eip712
+
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.hexAsBigInteger
+import pm.gnosis.utils.hexStringToByteArray
+import java.io.InputStream
+import java.math.BigInteger
+
+data class DomainWithMessage(val signer: Solidity.Address, val domain: Struct712, val message: Struct712)
+
+class EIP712JsonParser(private val jsonAdapter: JsonAdapter) {
+    fun parseMessage(rawJson: String): DomainWithMessage = parseMessage(jsonAdapter.parse(rawJson))
+
+    fun parseMessage(inputStream: InputStream): DomainWithMessage = parseMessage(jsonAdapter.parse(inputStream))
+
+    private fun parseMessage(adapterResult: JsonAdapter.Result): DomainWithMessage {
+        val primaryTypeParams = adapterResult.types[adapterResult.primaryType]
+            ?: throw IllegalArgumentException("TypedData does not contain primary type ${adapterResult.primaryType}")
+
+        val domainTypeParams = adapterResult.types[DOMAIN_TYPE]
+            ?: throw IllegalArgumentException("TypedData does not contain a domain")
+
+        return DomainWithMessage(
+            signer = adapterResult.signer,
```

Comment:
For me this doesn't belong here

Reply:
This was added later indeed. I will probably change the name of the data class (it should represent the result of the parsing with all the information required to sign)

Reply:
Fixed in https://github.com/gnosis/svalinn-kotlin/pull/31/commits/10cc16042762e85f3bc4b6a25f4241cec57d8a94